code/game/turfs/simulated/floor_damage.dm

```diff
@@ -50,8 +50,11 @@
 	if (health <= 0)
 		//Leftover damage will carry over to whatever tile replaces this one
 		var/leftover = abs(health)
+		//Small sanity to prevent -inf, also prevents recusively destorying tiles that (note that if floor tiles ever get less then 100 HP)
+		if(leftover >= 100)
```

Comment:
There should be a function of some sort that limits the damage, but it won't stop the infinite loop from happening. I'd say something like `leftover = min(leftover, 10 * max_health)` or something similar so that damage doesn't just vanish if it is actual high damage that it's receiving.